psiturk/experiment.py

```diff
@@ -263,16 +263,21 @@ def advertisement():
         # them to start the task again.
         raise ExperimentError('already_started_exp_mturk')
     elif status == COMPLETED:
-        # They've done the debriefing but perhaps haven't submitted the HIT
+        # They've finished the experiment but perhaps haven't submitted the HIT
         # yet.. Turn asignmentId into original assignment id before sending it
         # back to AMT
-        return render_template(
-            'thanks.html',
-            is_sandbox=(mode == "sandbox"),
-            hitid=hit_id,
-            assignmentid=assignment_id,
-            workerid=worker_id
-        )
+        use_psiturk_ad_server = CONFIG.getboolean('Shell Parameters', 'use_psiturk_ad_server')
+        if not use_psiturk_ad_server:
+            return render_template(
+                'thanks-mturksubmit.html',
+                using_sandbox=(mode == "sandbox"),
+                hitid=hit_id,
+                assignmentid=assignment_id,
+                workerid=worker_id
+            )
+        else: 
+            # then AMT isn't involved. Show them a thanks message and tell them to go away
+            return render_template( 'thanks.html' )
```

Comment:
@jhamrick I admit that I'm a bit confused about the purpose of `thanks.html`. It was [put back in 2.1.2](https://github.com/NYUCCL/psiTurk/commit/8195852822d17c902ab72608bebe28d7039d6f5a) by @gureckis. It doesn't use `is_sandbox`, `hitid`, `assignmentid`, or `workerid` anymore (it used to, before psiturk 2.0. It used to have the "click here to submit your hit" button). I couldn't figure out a way how workers would ever get to this page if they were using AMT. If they try to revisit the ad page, AMT just says "no more hits available to you in this group" and doesn't show the ad... at least that's how it went for me. Todd said this in the commit: "local thanks page needed when reloading the local ad and a worker has already completed the task." 

The only time I could see it being reached is if it were in debug (non-AMT) mode and a worker re-visited the /ad page. I guess though that /ad wouldn't be involved in debug mode.

My confusion aside, it should all still work fine though, just removing the "then AMT isn't involved" part.